pkg/api/mlflow/dao/repositories/metric.go

```diff
@@ -82,42 +82,24 @@ func (r MetricRepository) CreateBatch(
 
 	lastIters := make(map[string]int64)
 	for _, lastMetric := range lastMetrics {
-		lastIters[lastMetric.Key] = lastMetric.LastIter
+		lastIters[lastMetric.UniqueKey()] = lastMetric.LastIter
 	}
-	// uniqueContexts potentially up to length of metrics but most likely far less
+	allContexts := make([]*models.Context, len(metrics))
 	uniqueContexts := make([]*models.Context, 0, len(metrics))
-	contextMap := make(map[string]*models.Context)
+	contextProcessed := make(map[string]*models.Context)
 	latestMetrics := make(map[string]models.LatestMetric)
-	for n, metric := range metrics {
-		if metric.Context != nil {
-			hash := getJsonHash(metric.Context.Json)
-			foundCtx := contextMap[hash]
-			if foundCtx == nil {
-				contextMap[hash] = metric.Context
-				uniqueContexts = append(uniqueContexts, metric.Context)
-			} else {
-				metrics[n].Context = foundCtx
-			}
-		}
-		metrics[n].Iter = lastIters[metric.Key] + 1
-		lastIters[metric.Key] = metrics[n].Iter
-		lm, ok := latestMetrics[metric.Key]
-		if !ok ||
-			metric.Step > lm.Step ||
-			(metric.Step == lm.Step && metric.Timestamp > lm.Timestamp) ||
-			(metric.Step == lm.Step && metric.Timestamp == lm.Timestamp && metric.Value > lm.Value) {
-			latestMetrics[metric.Key] = models.LatestMetric{
-				RunID:     metric.RunID,
-				Key:       metric.Key,
-				Value:     metric.Value,
-				Timestamp: metric.Timestamp,
-				Step:      metric.Step,
-				IsNan:     metric.IsNan,
-				LastIter:  metric.Iter,
-				Context:   metrics[n].Context,
-			}
+	for n := range metrics {
+		ctxHash := getJsonHash(metrics[n].Context.Json)
```

Comment:
just a tiny pick -> maybe we can do `getJsonHash` function part of `Context` object?

Reply:
ok